sql/alps_submitter.go

```diff
@@ -0,0 +1,167 @@
+package sql
+
+import (
+	"fmt"
+	"io"
+	"strconv"
+)
+
+type GraphConfig struct {
+	TableName string
+	Fields    []string
+	X         []FeatureSpec
+	Y         FeatureSpec
+}
+
+type FeatureSpec struct {
+	FeatureName string
+	IsSparse    bool
+	Shape       []int
+	DType       string
+}
+
+type NumericColumn struct {
+	Key   string
+	Shape int
+}
+
+type BucketColumn struct {
+	SourceColumn NumericColumn
+	Boundaries   []int
+}
+
+type CrossColumn struct {
+	Keys           []interface{}
+	HashBucketSize int
+}
+
+func resolveFeatureSpec(el *exprlist) (*FeatureSpec, error) {
+	headExpr := (*el)[0].val
+	if headExpr == "DENSE" {
+		shape, err := strconv.Atoi((*el)[2].val)
```

Comment:
Check the length of `el`, `(*el)[2]`will panic if a user writes `DENSE(c2)`.

Reply:
Yes, just added a commit for this kind of situation, please take a review. @tonyyang-svail